packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -254,79 +268,83 @@ const helperBehavior = {
       return;
     }
 
-    const liquidationMargin = state.factoryPowers
-      .getGovernedParams()
-      .getLiquidationMargin();
-
-    // ask to be alerted when the price level falls enough that the vault
-    // with the highest debt to collateral ratio will no longer be valued at the
-    // liquidationMargin above its debt.
-    const triggerPoint = ceilMultiplyBy(
-      highestDebtRatio.numerator, // debt
-      liquidationMargin,
-    );
+    // INTERLOCK: the first time through, start the activity to wait for
+    // and process liquidations over time.
+    if (!liquidationInProgress) {
+      liquidationInProgress = true;
+      // eslint-disable-next-line consistent-return
+      return facets.helper
+        .processLiquidations()
+        .catch(e => console.log('Liquidator failed', e))
+        .finally(() => {
+          liquidationInProgress = false;
+        });
+    }
 
-    // INTERLOCK NOTE:
-    // The presence of an `outstandingQuote` is used to ensure there's only
-    // one thread of control waiting for a quote from the price authority.
-    // All later callers will just update the threshold for the `outstandingQuote`.
-    // The first invocation will skip the next block and continue to below where
-    // the `outstandingQuote` is assigned. So, if there's an outstanding quote,
-    // reset the level. If there's no current quote (because this is the first
-    // vault, or because a quote just resolved), then make a new request to the
-    // priceAuthority, and when it resolves, liquidate the first vault.
-    if (state.outstandingQuote) {
+    // There is already an activity processing liquidations. It may be
+    // waiting for the oracle price to cross a threshold.
+    // Update the current in-progress quote if there is one. Otherwise,
+    // the new threshold will be picked up by the next quote request.
+    if (outstandingQuote) {
```

Comment:
don't know if you saw my last comment here (not being passive aggressive ;-), just thought it might be lost in the rebasing )

This would let the "actually reschedule price check" be unnested and be consistent with the `if (!liquidationInProgress) {` early return style:

```
if (outstandingQuote === null) {
  // Don't schedule. The new threshold will be picked up by the next quote request.
  return;
}
```

Reply:
I didn't see the last comment previously, but saw it when looking at these comments. Addressed.